### PR TITLE
vulnerability:feature - Add new management of vulnerabilities back

### PR DESCRIPTION
**Pull request by wiliansilvazup**
Adding new routes for get files vulnerables and your total.
Updating route for get vulnerabilities by file, this file now is required in this routes.
The Param vulnType on query string of both routes now is required.

Signed-off-by: wilian <wilian.silva@zup.com.br>

### CODE REVIEW

File: vulnerability/internal/controllers/management/management.go [hunk 1]
```diff
@@ -31,8 +33,10 @@ import (
 	managementUseCases "github.com/ZupIT/horusec-platform/vulnerability/internal/usecase/management"
 )
 
+// nolint:lll // is not necessary broken interface
 type IController interface {
-	GetAllVulnerabilities(filter *managementEntities.Filter) (*managementEntities.Response, error)
+	GetAllVulnerabilitiesOfTheFile(filter *managementEntities.Filter) (*managementEntities.ResponseVulnerabilitiesOfTheFile, error)
```
Comment (nathanmartinszup) [inline vulnerability/internal/controllers/management/management.go:38]: Change `GetAllVulnerabilitiesOfTheFile` to `ListVulnerabilitiesByFile`

File: vulnerability/internal/controllers/management/management.go [hunk 2]
```diff
@@ -31,8 +33,10 @@ import (
...
+	GetAllFilesVulnerable(filter *managementEntities.Filter) (*managementEntities.ResponseFilesVulnerable, error)
```
Comment (nathanmartinszup) [inline vulnerability/internal/controllers/management/management.go:39]: Change `GetAllFilesVulnerable` to `ListVulnerableFiles`

File: vulnerability/internal/entities/management/response.go [hunk 3]
```diff
@@ -14,7 +14,27 @@
...
-	TotalItems int             `json:"totalItems"`
-	Data       *[]ResponseData `json:"data"`
+import (
+	"time"
+
+	vulnerabilityEntities "github.com/ZupIT/horusec-devkit/pkg/entities/vulnerability"
+	"github.com/google/uuid"
+)
+
+type ResponseVulnerabilitiesOfTheFile struct {
```
Comment (nathanmartinszup) [inline vulnerability/internal/entities/management/response.go:24]: Change to `ResponseVulnerabilitiesByFile`

File: vulnerability/internal/entities/management/response.go [hunk 4]
```diff
@@ -14,7 +14,27 @@
...
+	TotalItems int                                    `json:"totalItems"`
+	Data       *[]vulnerabilityEntities.Vulnerability `json:"data"`
+}
+
+type ResponseFilesVulnerable struct {
+	TotalItems int                            `json:"totalItems"`
+	Data       *[]ResponseDataFilesVulnerable `json:"data"`
+}
+
+type ResponseDataFilesVulnerable struct {
```
Comment (nathanmartinszup) [inline vulnerability/internal/entities/management/response.go:34]: Change to `ResponseVulnerableFiles`

File: vulnerability/internal/entities/management/response.go [hunk 5]
```diff
@@ -14,7 +14,27 @@
...
+import (
+	"time"
+
+	vulnerabilityEntities "github.com/ZupIT/horusec-devkit/pkg/entities/vulnerability"
+	"github.com/google/uuid"
+)
+
+type ResponseVulnerabilitiesOfTheFile struct {
+	TotalItems int                                    `json:"totalItems"`
+	Data       *[]vulnerabilityEntities.Vulnerability `json:"data"`
```
Comment (nathanmartinszup) [inline vulnerability/internal/entities/management/response.go:26]: I guess this pointer is unnecessary

File: vulnerability/internal/entities/management/response.go [hunk 6]
```diff
@@ -14,7 +14,27 @@
...
+}
+
+type ResponseFilesVulnerable struct {
+	TotalItems int                            `json:"totalItems"`
+	Data       *[]ResponseDataFilesVulnerable `json:"data"`
```
Comment (nathanmartinszup) [inline vulnerability/internal/entities/management/response.go:31]: Same here

File: vulnerability/internal/entities/management/filter.go [hunk 7]
```diff
@@ -92,10 +99,18 @@ func (f *Filter) setRepositoryIDFromRequest(r *http.Request) error {
 	return nil
 }
 
-func (f *Filter) setVulnerabilityFilters(r *http.Request) {
+func (f *Filter) setVulnerabilityFilters(r *http.Request, validateVulnFile bool) error {
```
Comment (nathanmartinszup) [inline vulnerability/internal/entities/management/filter.go:102]: This validations should be in the `Validate` func

Comment (wiliansilvazup) [reply vulnerability/internal/entities/management/filter.go:102]: done

**Review by nathanmartinszup — APPROVED** [review]

**Review by lucasbrunozup — APPROVED** [review]

**Pull request merged**